src/OpenTelemetry.Api/Context/Propagation/TraceContextFormat.cs

```diff
@@ -304,5 +283,25 @@ private bool TryExtractTracestate(string[] tracestateCollection, out string trac
 
             return true;
         }
+
+        private static byte HexCharToByte(char c)
+        {
+            if ((c >= '0') && (c <= '9'))
+            {
+                return (byte)(c - '0');
```

Comment:
`return Convert.ToByte(c)` would work better here than doing inline subtractions / additions.

Same for upper and lower casings below.

Reply:
Can you validate?